libs/vm/src/parser.cpp

```diff
@@ -1024,304 +1016,172 @@ ExpressionNodePtr Parser::ParseExpression(bool is_conditional_expression)
-      {
-        return nullptr;
-      }
+      parses = HandleBinaryOp(NodeKind::LessThan, OpInfo(4, Association::Left, 2));
       break;
-    }
+
     case Token::Kind::LessThanOrEqual:
-    {
-      if (HandleBinaryOp(NodeKind::LessThanOrEqual, OpInfo(4, Association::Left, 2)) == false)
-      {
-        return nullptr;
-      }
+      parses = HandleBinaryOp(NodeKind::LessThanOrEqual, OpInfo(4, Association::Left, 2));
       break;
-    }
+
     case Token::Kind::GreaterThan:
-    {
-      if (HandleBinaryOp(NodeKind::GreaterThan, OpInfo(4, Association::Left, 2)) == false)
-      {
-        return nullptr;
-      }
+      parses = HandleBinaryOp(NodeKind::GreaterThan, OpInfo(4, Association::Left, 2));
       break;
-    }
+
     case Token::Kind::GreaterThanOrEqual:
-    {
-      if (HandleBinaryOp(NodeKind::GreaterThanOrEqual, OpInfo(4, Association::Left, 2)) == false)
-      {
-        return nullptr;
-      }
+      parses = HandleBinaryOp(NodeKind::GreaterThanOrEqual, OpInfo(4, Association::Left, 2));
       break;
-    }
+
     case Token::Kind::And:
-    {
-      if (HandleBinaryOp(NodeKind::And, OpInfo(2, Association::Left, 2)) == false)
-      {
-        return nullptr;
-      }
+      parses = HandleBinaryOp(NodeKind::And, OpInfo(2, Association::Left, 2));
       break;
-    }
+
     case Token::Kind::Or:
-    {
-      if (HandleBinaryOp(NodeKind::Or, OpInfo(1, Association::Left, 2)) == false)
-      {
-        return nullptr;
-      }
+      parses = HandleBinaryOp(NodeKind::Or, OpInfo(1, Association::Left, 2));
       break;
-    }
+
     case Token::Kind::Not:
-    {
       HandleNot();
       break;
-    }
+
     case Token::Kind::Inc:
-    {
       HandlePrefixPostfix(NodeKind::PrefixInc, OpInfo(7, Association::Right, 1),
                           NodeKind::PostfixInc, OpInfo(8, Association::Left, 1));
       break;
-    }
+
     case Token::Kind::Dec:
-    {
       HandlePrefixPostfix(NodeKind::PrefixDec, OpInfo(7, Association::Right, 1),
                           NodeKind::PostfixDec, OpInfo(8, Association::Left, 1));
       break;
-    }
+
     case Token::Kind::LeftParenthesis:
-    {
-      if (HandleOpener(NodeKind::ParenthesisGroup, NodeKind::Invoke, Token::Kind::RightParenthesis,
-                       ")") == false)
-      {
-        return nullptr;
-      }
+      parses = HandleOpener(NodeKind::ParenthesisGroup, NodeKind::Invoke,
+                            Token::Kind::RightParenthesis, ")");
       break;
-    }
+
     case Token::Kind::LeftSquareBracket:
-    {
-      if (HandleOpener(NodeKind::Unknown, NodeKind::Index, Token::Kind::RightSquareBracket, "]") ==
-          false)
-      {
-        return nullptr;
-      }
+      parses =
+          HandleOpener(NodeKind::Unknown, NodeKind::Index, Token::Kind::RightSquareBracket, "]");
       break;
-    }
+
+    case Token::Kind::LeftBrace:
+      parses = HandleOpener(NodeKind::Unknown, NodeKind::Index, Token::Kind::RightBrace, "}");
```

Comment:
This is wrong. It should be HandleOpener(NodeKind::InitializerList, NodeKind::Unknown....

Reply:
Thanks.